pkg/dialer/factory.go

```diff
@@ -54,15 +58,24 @@ func (f *Factory) ClusterDialer(clusterName string) (dialer.Dialer, error) {
 	}, nil
 }
 
-func IsCloudDriver(cluster *v3.Cluster) bool {
-	return !cluster.Spec.Internal &&
-		cluster.Status.Driver != "" &&
-		cluster.Status.Driver != v32.ClusterDriverImported &&
-		cluster.Status.Driver != v32.ClusterDriverRKE &&
-		cluster.Status.Driver != v32.ClusterDriverK3s &&
-		cluster.Status.Driver != v32.ClusterDriverK3os &&
-		cluster.Status.Driver != v32.ClusterDriverRke2 &&
-		cluster.Status.Driver != v32.ClusterDriverRancherD
+func IsPublicCloudDriver(cluster *v3.Cluster) bool {
+	if cluster.Spec.Internal ||
+		cluster.Status.Driver == "" ||
+		cluster.Status.Driver == v32.ClusterDriverImported ||
+		cluster.Status.Driver == v32.ClusterDriverRKE ||
+		cluster.Status.Driver == v32.ClusterDriverK3s ||
+		cluster.Status.Driver == v32.ClusterDriverK3os ||
+		cluster.Status.Driver == v32.ClusterDriverRke2 ||
+		cluster.Status.Driver == v32.ClusterDriverRancherD {
+		return false
+	}
+
+	return !IsPrivateCloudDriver(cluster)
+}
+
+func IsPrivateCloudDriver(cluster *v3.Cluster) bool {
```

Comment:
I think the name is a little confusing on this and IsPublicCloudDriver. A name like IsPrivateCloudCluster, and possibly a comment to explain this refers to a cluster with only private access opposed to one that has at least private access.

Reply:
Will update.